pyproject.toml

```diff
@@ -303,3 +304,57 @@ exclude_lines = [
     'if __name__ == .__main__.:',
 ]
 ignore_errors = true
+
+[tool.pyright]
```

Comment:
Is there a `pyright.toml` file we could use akin to `ruff.toml`?

Reply:
Not as far as I'm aware. FWIW Sphinx uses `pyproject.toml` for `ruff` and `mypy` configuration. There is `pyrightconfig.json`.

Reply:
We use ``.ruff.toml`` -- I prefer using dedicated ``.toml`` files per tool when possible, as ``pyproject.toml`` is far too large.

Reply:
It seems we could use ``pyrightconfig.json``, but that seems a bad option as JSON files are challenging to work with by hand.

Reply:
Oh yes!

I think in the future it might be good to switch to `pyrightconfig.json` as that allows for more precise configuration.